runtime/pkg/logutil/handler.go

```diff
@@ -1,45 +1,41 @@
 package logutil
 
 import (
-	"context"
-
 	"github.com/rilldata/rill/runtime/pkg/logbuffer"
-	"go.uber.org/zap/exp/zapslog"
-	"golang.org/x/exp/slog"
+	"go.uber.org/zap/zapcore"
 )
 
-type DuplicatingHandler struct {
-	zapHandler *zapslog.Handler
-	logs       *logbuffer.Buffer
+type BufferedZapCore struct {
+	fields []zapcore.Field
+	logs   *logbuffer.Buffer
 }
 
-func NewDuplicatingHandler(zapHandler *zapslog.Handler, logs *logbuffer.Buffer) *DuplicatingHandler {
-	return &DuplicatingHandler{
-		zapHandler: zapHandler,
-		logs:       logs,
+func NewBufferedZapCore(logs *logbuffer.Buffer) *BufferedZapCore {
+	return &BufferedZapCore{
+		logs: logs,
 	}
 }
 
-func (d *DuplicatingHandler) Enabled(ctx context.Context, level slog.Level) bool {
-	return d.zapHandler.Enabled(ctx, level)
+func (d *BufferedZapCore) Enabled(level zapcore.Level) bool {
+	return true
 }
 
-func (d *DuplicatingHandler) Handle(ctx context.Context, record slog.Record) error {
-	err := d.zapHandler.Handle(ctx, record)
-	if err != nil {
-		return err
-	}
-	return d.logs.Add(record)
+func (d *BufferedZapCore) Write(entry zapcore.Entry, fields []zapcore.Field) error {
+	return d.logs.AddZapEntry(entry, fields)
+}
+
+func (d *BufferedZapCore) With(fields []zapcore.Field) zapcore.Core {
+	// TODO: should ideally create a clone of the core with new log buffer and append the new fields
+	clone := *d
+	clone.fields = append(clone.fields, fields...)
+	return &clone
```

Comment:
1. May be I was just confused that `With` should return a new independent logger but that does not seem like it.
2.  Not sure I understand how it will overwrite as `append` adds to the end

Reply:
For 2 I understand what you mean - the fields for the original logger will also get the appended fields so we need to create a new slice so original array remains as it is.

Reply:
Actually it's a little more complicated – see this example: https://go.dev/play/p/ghSHN5YV9Du.

Basically slices represent ranges on top of an underlying allocated array. So while `With` won't change the slice of the original, it can corrupt the slice of another logger previously created using `With`.

Reply:
ok interesting, thanks for the explanation.